### PR TITLE
Inherit render properties

### PR DESCRIPTION
**Pull request by BMaxV**
Hi, 

this is for #114 .

There are two things probably still have to be done: 

I don't know what good non default values would be for 'encoding' and 'engine' for the test function.

There is a check in https://github.com/xflr6/graphviz/blob/master/graphviz/dot.py#L100 that subgraphs can't be strict, so it does not make sense to inherit that one.

### CODE REVIEW

File: graphviz/dot.py [hunk 1]
```diff
@@ -67,6 +67,7 @@ def __init__(self, name=None, comment=None,
         self.body = list(body) if body is not None else []
 
         self.strict = strict
+        self.enconding = encoding
```
Comment (xflr6) [inline graphviz/dot.py:70]: This is handled by the `super()`-call in line 61 above:

https://github.com/xflr6/graphviz/blob/0c503a32f5f17f99a6e0ad8ac29fb441e3c0a9f2/graphviz/files.py#L103

**Comment by xflr6** [comment]
Thanks for working on this.

> I don't know what good non default values would be for 'encoding' and 'engine' for the test function.

How about `'ascii'`, and `'neato'`?

> There is a check in https://github.com/xflr6/graphviz/blob/master/graphviz/dot.py#L100 that subgraphs can't be strict, so it does not make sense to inherit that one.

https://github.com/xflr6/graphviz/blob/0c503a32f5f17f99a6e0ad8ac29fb441e3c0a9f2/graphviz/dot.py#L98-L100

+1. We should add a note to the `.subgraph()`-method that instances created with the context manager are always `strict=None` because the parent's value is used when rendering the parent graph (there is no way for the subgraph to override). This means if we render the subgraph in the context manager it is `bool(strict) == False`. If user's would really want to render the subgraph as strict while rendering the whole graph non-strict they could do like this.

```python
(...)
with d.subgraph(name='child') as s:
    s.strict = True
    s.render('child.gv')
    s.strict = None  # any falcy value
```

I think we should add a functional test for the use case of doing `render()` on a subgraph within the context-manager, i.e. use `tmpdir` and assert on the rendered files. Not yet shure where to put it exactly.

Another thing would be to extend the the docstring and maybe also the [documentation section](https://graphviz.readthedocs.io/en/stable/manual.html#subgraphs-clusters) and to write a short summary of the behavioural difference for the CHANGES.txt. Tell me if/when you prefer me to take over, then we can probably merge what you have.

**Comment by BMaxV** [comment]
I think this should be it. I didn't touch CHANGES.txt yet , that would still need to be done. It inherits the format now and that was the main thing I wanted, so I think you can take over now?

**Comment by xflr6** [comment]
Thanks. Still need a functional test, implement `strict=None`, more informative documentation and a proper note in the docstring, let me take if from here then.

**Comment by xflr6** [comment]
See adaptations in 4a20d2e4183aa2ac99c07706d2dfd6803f054cca
